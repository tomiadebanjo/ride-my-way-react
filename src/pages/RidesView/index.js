import React from 'react';
import { connect } from 'react-redux'
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import RideCard from '../../components/RideCard';
import classes from './style.scss';
import { fetchRide } from '../../store/actions/ride';

class RidesView extends React.Component {
  componentDidMount(){
    const { fetchRideNow } = this.props
    fetchRideNow();
  }

  render(){
    const { rides, loading } = this.props;
    const mappedRides = rides.map(ride => {
      const {
        id,
        destination,
        pickup_location,
        departure_date,
        departure_time
      } = ride
      return (
        <RideCard
          key={id}
          destination={destination}
          pickup_location={pickup_location}
          departure_date={departure_date}
          departure_time={departure_time}
        />
      )
    })
    return (
      <div className={classes.ride_container}>
        <Header />
        <div className={classes.main_container}>
          {
            loading ? <div className={classes.loader}></div> : mappedRides
          }
        </div>
        <Footer />
      </div>
    )
  }
};

const mapStateToProps = state => ({
	loading: state.ride.fetchRides.loading,
	rides: state.ride.fetchRides.rides
});

const mapDispatchToProps = dispatch => ({
  fetchRideNow: () => dispatch(fetchRide())
})

export default connect(mapStateToProps, mapDispatchToProps)(RidesView);
