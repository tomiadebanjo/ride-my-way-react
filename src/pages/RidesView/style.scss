.ride_container {
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  .heading {
    text-align: center;
    font-weight: 400;
    margin-bottom: 0;
  }

  .main_container {
    display: inline-grid;
    flex: 1 ;
    flex-wrap: wrap;
    align-content: flex-start;
    grid-template-columns: auto auto auto auto;
    grid-gap: 20px;
    padding: 20px;

    .loader {
      display: block;
      margin: auto;
      border: 10px solid #f3f3f3;
      border-top: 10px solid #3498db;
      border-radius: 50%;
      width: 120px;
      height: 120px;
      position: fixed;
      top: 40vh;
      left: 45%;
      animation: spin 2s linear infinite;
      -webkit-animation: spin 2s linear infinite;
    }

    @keyframes spin {
      0% { transform: rotate(0deg); }
      100% { transform: rotate(360deg); }
    }
  }
}

@media (max-width: 1024px){
  .ride_container {
    .main_container {
      grid-template-columns: auto auto auto
    }
  }
}

@media (max-width: 768px){
  .ride_container {
    .main_container {
      grid-template-columns: auto auto
    }
  }
}

@media (max-width: 475px){
  .ride_container {
    .main_container {
      grid-template-columns: auto
    }
  }
}