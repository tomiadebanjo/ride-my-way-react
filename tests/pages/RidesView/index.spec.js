import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import RidesView from '../../../src/pages/RidesView';

const middleware = [thunk];
const mockStore = configureStore(middleware);
let wrapper;
let store;

const initialState = {
	authentication: {
		isAuthenticated: false
	},
	ride: {
		fetchRides: {
			loading: false,
			rides: [
				{
					departure_time: 'time',
					departure_date: 'date',
					pickup_location: 'location',
          destination: 'destination',
          id: 1
				}
			]
		}
	}
};

describe('ride view page test', () => {
	beforeEach(() => {
		store = mockStore(initialState);
		wrapper = mount(
			<Provider store={store}>
				<BrowserRouter>
					<RidesView />
				</BrowserRouter>
			</Provider>
		);
	});

  it('should render the ride view page  correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
