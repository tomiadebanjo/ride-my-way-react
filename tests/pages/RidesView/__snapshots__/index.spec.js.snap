// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ride view page test should render the ride view page  correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(RidesView)>
        <RidesView
          fetchRideNow={[Function]}
          getSingleRide={[Function]}
          loading={false}
          rides={
            Array [
              Object {
                "departure_date": "date",
                "departure_time": "time",
                "destination": "destination",
                "id": 1,
                "pickup_location": "location",
              },
            ]
          }
        >
          <div
            className="ride_container"
          >
            <Connect(Header)>
              <Header
                dispatch={[Function]}
                isAuthenticated={false}
              >
                <header
                  className="header"
                >
                  <a
                    className="brand"
                    href="/"
                  >
                    Ride My Way
                  </a>
                  <nav>
                    <ul
                      className="menu"
                    >
                      <li>
                        <Link
                          replace={false}
                          to="/login"
                        >
                          <a
                            href="/login"
                            onClick={[Function]}
                          >
                            Login
                          </a>
                        </Link>
                      </li>
                      <li>
                        <Link
                          replace={false}
                          to="/signup"
                        >
                          <a
                            href="/signup"
                            onClick={[Function]}
                          >
                            Sign-Up
                          </a>
                        </Link>
                      </li>
                    </ul>
                  </nav>
                </header>
              </Header>
            </Connect(Header)>
            <SingleRideModal
              handleClose={[Function]}
              show={false}
            >
              <Modal
                handleClose={[Function]}
                show={false}
              >
                <div
                  className="modal  display_none"
                >
                  <section
                    className="modal_main"
                  >
                    <span
                      className="close"
                      id="ride-close"
                      onClick={[Function]}
                    >
                      ×
                    </span>
                    <div
                      className="single_ride_modal"
                    >
                      <h2>
                        Ride Details
                      </h2>
                      <div
                        className="loader"
                      />
                    </div>
                  </section>
                </div>
              </Modal>
            </SingleRideModal>
            <h2
              className="heading"
            >
              Available rides
            </h2>
            <div
              className="main_container"
            >
              <RideCard
                departure_date="date"
                departure_time="time"
                destination="destination"
                key="1"
                pickup_location="location"
                show={[Function]}
              >
                <div
                  className="card"
                >
                  <div
                    className="location_row"
                  >
                    <div
                      className="location_text"
                    >
                      <p>
                        location
                      </p>
                    </div>
                    <div
                      className="img_size"
                    >
                      <img
                        alt="arrow"
                        className="img"
                        src="test-file-stub"
                      />
                    </div>
                    <div
                      className="location_text"
                    >
                      <p>
                        destination
                      </p>
                    </div>
                  </div>
                  <div
                    className="time"
                  >
                    <p>
                      time
                       / 
                      Invalid Date
                    </p>
                  </div>
                  <div
                    className="button"
                  >
                    <a
                      onClick={[Function]}
                    >
                      More Info
                    </a>
                  </div>
                </div>
              </RideCard>
            </div>
            <Footer>
              <footer>
                <p>
                  Ride My Way | Copyright © 2018
                </p>
              </footer>
            </Footer>
          </div>
        </RidesView>
      </Connect(RidesView)>
    </Router>
  </BrowserRouter>
</Provider>
`;
